tests/phpunit/tests/test-wpml-config-locations.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+class WPML_Config_Locations_Test extends PLL_UnitTestCase {
+
+	protected static $dirs = array(
+		'themes' => array(
+			'best-theme',
+			'best-child',
+		),
+		'plugins' => array(
+			'best-plugin',
+		),
+	);
+
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		parent::wpSetUpBeforeClass( $factory );
+
+		// Copy themes and plugins from tests/phpunit/data to wp-content.
+		foreach ( self::$dirs as $type => $subdirs ) {
+			foreach ( $subdirs as $name ) {
+				$source_dir = dirname( __DIR__ ) . "/data/{$type}/{$name}";
+				$dest_dir   = WP_CONTENT_DIR . "/{$type}/{$name}";
+
+				@mkdir( $dest_dir );
+
+				foreach ( glob( $source_dir . '/*.*' ) as $file ) {
+					copy( $file, $dest_dir . '/' . basename( $file ) );
+				}
+			}
+		}
+	}
+
+	public static function wpTearDownAfterClass() {
+		parent::wpTearDownAfterClass();
+
+		// Remove previously copied themes and plugins from wp-content.
+		foreach ( self::$dirs as $type => $subdirs ) {
+			foreach ( $subdirs as $name ) {
+				$dest_dir = WP_CONTENT_DIR . "/{$type}/{$name}";
+
+				foreach ( glob( $dest_dir . '/*.*' ) as $file ) {
+					unlink( $file );
+				}
+
+				rmdir( $dest_dir );
+			}
+		}
+	}
+
+	public function test_in_polylang() {
+		@mkdir( WP_CONTENT_DIR . '/polylang' );
+		copy( dirname( __DIR__ ) . '/data/wpml-config.xml', WP_CONTENT_DIR . '/polylang/wpml-config.xml' );
+
+		$files    = ( new PLL_WPML_Config() )->get_files();
+		$expected = array( 'Polylang' => WP_CONTENT_DIR . '/polylang/wpml-config.xml' );
+
+		unlink( WP_CONTENT_DIR . '/polylang/wpml-config.xml' );
+		rmdir( WP_CONTENT_DIR . '/polylang' );
```

Comment:
This should be moved to `wpTearDownAfterClass()` with some `file_exist()` IMHO.

Reply:
Not in `wpTearDownAfterClass()` as this would pollute other tests.
It could be moved to `tear_down()` but I prefer to clean up in the test when it's possible to have everything in one single place. Sometimes it's not desirable to avoid duplicate code, or not possible. But here it is as the assertion can be done after the clean up.